docs/type.md

```diff
@@ -1,98 +1,197 @@
 ```
-<!--Output-->
-```js
-var score = 10;
+
+## Aliases
+
+Aliases can be defined for types. This is helpful to attach meaning to simple
+types and when working with complex types that become long to write down.
+
+```reason
+type seconds = int;
+type timeInterval = (seconds, seconds);
 ```
-<!--END_DOCUSAURUS_CODE_TABS-->
 
-You can also wrap any expression in parentheses and annotate it:
+Using the alias `seconds` it is clear how the `sleep` function works. If `int`
+were used it might not be obvious:
 
-<!--DOCUSAURUS_CODE_TABS-->
-<!--Reason-->
 ```reason
-let myInt1 = 5;
-let myInt2: int = 5;
-let myInt3 = (5: int) + (4: int);
-let add = (x: int, y: int): int => x + y;
-let drawCircle = (~radius as r: int) : string => "hi";
+let sleep = (time: seconds) => { ... }
 ```
-<!--Output-->
-```js
-var myInt1 = 5;
-var myInt2 = 5;
-var myInt3 = 9;
-function addInts(x, y) {
-  return x + y | 0;
-}
-function drawCircle(r) {
-  return "hi";
-}
+
+Type aliases are required in some cases, such as working with
+[variants](variant.md) or [records](record.md).
+
+## Type Arguments
+
+Types can have arguments. This is useful when defining structures that work
+with many types of values. Having a single `list` type with an argument that can
+be `int`, `float`, or `string`, is better than having three concrete `intList`,
+`floatList`, and `stringList` types.
+
+Arguments are prefixed with a single `'` when defining the type:
+
+```reason
+type list('item) = ...
 ```
-<!--END_DOCUSAURUS_CODE_TABS-->
 
-Note: in the last line, `(~radius as r: int)` is a labeled argument. More on this [here](function.md).
+When using this type as an annotation the argument must be filled in with a
+concrete type:
 
-## Aliases
+```reason
+let x: list(int) = [1, 2, 3];
+let y: list(string) = ["one", "two", "three"];
+```
 
-You can refer to a type by a different name. They'll be equivalent:
+Types can have multiple arguments and be nested:
 
-<!--DOCUSAURUS_CODE_TABS-->
-<!--Reason-->
 ```reason
-type scoreType = int;
-let x: scoreType = 10;
+type pair('a, 'b) = ('a, 'b);
 
+let x: pair(int, string) = (1, "one");
+let y: pair(string, list(int)) = ("123", [1, 2, 3]);
 ```
-<!--Output-->
-```js
-var score = 10;
+
+- Note: It is common convention for type arguments to be named `'a`, `'b`,
+`'c`, etc.
+- Note: Type arguments can still be inferred!
+
+## Opaque Types
+
+Opaque types are a powerful tool that limits the implementation details that
+are exposed to users. This makes it easier and safer to modify implementations
+to suit changing needs.
+
+_Note: In these examples module types are used for simplicity, but opqaue types
+are more often created using interface files._
+
+### Setup
+
+`Duration.t` is an opaque type. Its concrete type is hidden and it can only be
+interacted with using a limited set of functions:
+
+_(Writing `type t;` is what makes `t` opaque. `t` would not be opaque and would
+still have a concrete type if `type t = int;` were written instead.)_
+
+```reason
```

Comment:
I might just explain this in terms of interface files and implementation files since the reader has probably not covered modules yet. Then when modules are covered they can be explained in terms of files ("It's like embedding one Reason file inside of another").

Reply:
That's a lot harder to put into documentation and is more confusing IMO. At this point we also have not covered interface files, they will get covered at the same time as modules pretty much